src/NLog/LayoutRenderers/AllEventPropertiesLayoutRenderer.cs

```diff
@@ -137,23 +173,26 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
             if (logEvent.HasProperties)
             {
                 var formatProvider = GetFormatProvider(logEvent);
+                bool checkForExclude = CheckForExclude(logEvent);
+                bool nonStandardFormat = _beforeKey == null || _afterKey == null || _afterValue == null;
 
                 bool first = true;
-                IEnumerable<KeyValuePair<object, object>> properties = GetProperties(logEvent);
-                if (!IncludeEmptyValues)
-                {
-                    properties = properties.Where(p => !IsEmptyPropertyValue(p.Value));
-                }
-                foreach (var property in properties)
+                foreach (var property in logEvent.Properties)
                 {
+                    if (!IncludeEmptyValues && IsEmptyPropertyValue(property.Value))
+                        continue;
+
+                    if (checkForExclude && Exclude.Contains(property.Key as string))
```

Comment:
Proposal: 

```
  if (checkForExclude && property.Key is string key && Exclude.Contains(key))
```
saves as lookup when ` property.Key` is `null`

Reply:
Let me know what you think, i'm also fine with keeping it like this. Then I will merge it.

Reply:
Never thought you would suggest premature optimization, and increase code complexity. I can never say no to an optimization opportunity, even thought I think it will only help 1 out of 10 cases :)

Reply:
I was trigger by the `null` to Contains, which crashes on dictionaries (well on ContainsKey), and not really by the optimization ;)

Reply:
Optimization has been committed :)

Reply:
🎉